recipes/userspace-rcu/all/conanfile.py

```diff
@@ -0,0 +1,94 @@
+    description = "Userspace RCU (read-copy-update) library"
+    topics = ("conan", "urcu")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "LGPLv2.1"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    settings = "os", "compiler", "build_type", "arch"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "model": ["generic", "mb", "signal", "bp"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "model": "generic",
+    }
+    build_requires = (
+        "libtool/2.4.6",
+    )
+
+    def validate(self):
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            raise ConanInvalidConfiguration("Only Linux/FreeBSD supported")
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            self.run("./bootstrap")
+            tools.mkdir("_build")
+            with tools.chdir("_build"):
+                env_build = AutoToolsBuildEnvironment(self)
+                extra_args = list()
+                if self.options.shared:
+                    extra_args.append('--enable-static=no')
+                else:
+                    extra_args.append('--enable-shared=no')
+                env_build.configure("../", args=extra_args, build=False, host=False, target=False)
+                env_build.make()
+
+    def package(self):
+        self.copy(pattern="LICENSE*", src=self._source_subfolder, dst="licenses")
+        self.copy("*.h", excludes=("*_build/*", "*doc/*", "*tests/*"), src="{}/src".format(self._source_subfolder), dst="include/", keep_path=True)
+        self.copy("*.h", excludes=("*_build/*", "*doc/*", "*tests/*"), src="{}/include".format(self._source_subfolder), dst="include/", keep_path=True)
+        self.copy("*.h", src="{}/_build/include/urcu/".format(self._source_subfolder), dst="include/urcu", keep_path=False)
+        lib_name = ''
+        if self.options.model == "generic":
+            lib_name = 'liburcu'
+        elif self.options.model == "mb":
+            lib_name = 'liburcu-mb'
+        elif self.options.model == "signal":
+            lib_name = 'liburcu-signal'
+        else:
+            lib_name = 'liburcu-bp'
+        if self.options.shared:
+            self.copy("*{}.dll".format(lib_name), dst="bin", keep_path=False)
+            self.copy("*{}.so*".format(lib_name), dst="lib", keep_path=False)
+            self.copy("*{}.dylib".format(lib_name), dst="lib", keep_path=False)
+        else:
+            self.copy("*{}.a".format(lib_name), dst="lib", keep_path=False)
```

Comment:
Is there as reason that only on lib at a time is linked again but all of them are copied?

Reply:
I'm not sure I follow; this is the package() function so while all library variants are built, only the one selected with the `model` option is copied into the package (and thus added to the cpp_info linker flags). Am I missing something?

Reply:
We try to respect the original projects layout as closely as possible this way existing users of the project can continue what they are doing and new consumers can just follow the upstream docs without Conan getting in the way.

Looking of the project what we'd ideally have is the the projects install and model the `pkg_config` targets https://github.com/urcu/userspace-rcu/blob/7c40d7f5b2139ec9151daf8410b6b934234220d5/configure.ac#L310-L316

This was, as the original author intended, consumers can pick which libs they link against

***

My suggestion is 
- deleting the `model` option
- call the upstream install method (noted https://github.com/conan-io/conan-center-index/pull/11595#discussion_r924209798)
- add components (should be a one to one with the libs) [how to docs](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#autotools)

Reply:
Ah, I see. Thanks for the clarification!